### ISSUE TITLE
SA1009 false positive in range operator with parenthesis

### ISSUE THREAD

**Issue by Tan90909090** [DotNetAnalyzers/StyleCopAnalyzers#3064]
**Version Used**:  `StyleCop.Analyzers v1.2.0-beta.113`

**Steps to Reproduce**:
```cs
class C
{
    static void Main() { }

    void Foo(string x, int i)
    {
        _ = x[(i + 1)..]; // SA1009 is reported here
        _ = x[..(i + 1)]; // fixed with #3040
    }
}
```

**Expected Behavior**:
SA1008 false positive on `x[..(i + 1)]` have been fixed with #3040 so I think SA1009 should not be reported on `x[(i + 1)..]`.

**Actual Behavior**:
`SA1009: Closing parenthesis should be followed by a space` is reported on `x[(i + 1)..]`.

**Comment by watsonsong** [comment]
Is this issue resolved?
This error still report in '1.2.0-beta.354'

**Comment by sharwell** [comment]
@watsonsong We have a test for this here:

https://github.com/sharwell/StyleCopAnalyzers/blob/501b7c07fbadaab581d297cc9d7e348bd3527fce/StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1009CSharp8UnitTests.cs#L102-L147

It sounds like there may be a difference in the scenario where the warning is still reported. Can you suggest a new test?
